src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -508,6 +510,49 @@ public BoundStatementList StatementList(BoundStatement first, BoundStatement sec
             return new BoundStatementList(Syntax, ImmutableArray.Create(first, second)) { WasCompilerGenerated = true };
         }
 
+        [return: NotNullIfNotNull(nameof(first)), NotNullIfNotNull(nameof(second))]
+        public BoundStatement? Concat(BoundStatement? first, BoundStatement? second)
+            => (first == null) ? second : (second == null) ? first : StatementList(first, second);
+
+        public BoundBlockInstrumentation Instrumentation(BoundBlockInstrumentation? innerInstrumentation = null, LocalSymbol? local = null, BoundStatement? prologue = null, BoundStatement? epilogue = null)
+            => (innerInstrumentation != null)
```

Comment:
>=> [](http://example.com/codeflow?start=12&length=2)

Consider using regular block body. Given that the expression spans more than 10 lines of code, using an expression body doesn't provide any advantages. It actually negatively affects readability, in my opinion.